### PR TITLE
docs: add migration section for v2

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR adds a migration section to the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,28 +2,65 @@
 
 A JavaScript library for managing authentication for [Storyblok](https://www.storyblok.com) apps.
 
+## Migrating v1 to v2
+
+The `@storyblok/app-extension-auth` v1 stored the access token in a cookie. However, v2 now offers an adapter pattern, allowing you to store the access token elsewhere, such as in a dedicated database. And here are the breaking changes:
```
Comment (BibiSebi) [inline README.md:7]: I am wondering if this is a bit misleading, as we abstracted the logic to the adapter pattern however the storage inside a DB and enabling passing a custom adapter is not possible at the moment right?

Comment (eunjae-lee) [reply README.md:7]: I'm not sure if I follow you. How is it not possible?

Comment (BibiSebi) [reply README.md:7]: Currently the user cannot pass their own adapter to authHandler or the sessionStore.

Comment (eunjae-lee) [reply README.md:7]: Oh I didn't realize we removed adapter from AuthHandlerParams. I'll modify this.

File: README.md [hunk 2]
```diff
@@ -2,28 +2,65 @@
 
 A JavaScript library for managing authentication for [Storyblok](https://www.storyblok.com) apps.
 
+## Migrating v1 to v2
+
+The `@storyblok/app-extension-auth` v1 stored the access token in a cookie. However, in a future version of v2, the library will offer an adapter pattern, allowing you to store the access token elsewhere, such as in a dedicated database. And here are the breaking changes:
```
Comment (eunjae-lee) [inline README.md:7]: Not sure how to phrase this 🤔 what do you think?

Comment (BibiSebi) [reply README.md:7]: This is fine, maybe adding one more sentence to make it clearer, but its just nitpicking. 

```suggestion
The `@storyblok/app-extension-auth` v1 stored the access token in a cookie. However, in a future version of v2, the library will offer an adapter pattern, allowing you to store the access token elsewhere, such as in a dedicated database. The following changes abstract the code to make this transition easier in the near future:
```

**Review by BibiSebi — APPROVED** [review]

**Pull request merged**